api/samples/CMakeLists.txt

```diff
@@ -116,7 +116,7 @@ else (WIN32)
 endif (WIN32)
 
 if (DEBUG)
-  set(OPT_CFLAGS "-DDEBUG")
+  set(OPT_CFLAGS "${OPT_CFLAGS} -DDEBUG")
```

Comment:
Remember that samples/CMakeLists.txt is special with a dual use: it is also exported (after some edits) for use by users directly.  It seems best to add a set of DEBUG inside the `"${CMAKE_BUILD_TYPE}" MATCHES "Debug")` up top to match what cmake users will expect ("DEBUG" is not a standard cmake variable: the build type is the standard method).